src/main/resources/default/assets/scripts/list-layout-switch.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+function extendUrlWithDisplayMode(currentLocation, value) {
+    const key = "display-mode";
+    const url = new URL(currentLocation);
+    url.searchParams.delete(key);
+    url.searchParams.append(key, value);
+    return url.toString();
+}
+
+sirius.ready(function() {
+    document.querySelectorAll("a.list-link-cards-js").forEach(function (_link) {
+        _link.href = extendUrlWithDisplayMode(window.location.href, "CARDS");
+    });
+    document.querySelectorAll("a.list-link-list-js").forEach(function (_link) {
+        _link.href = extendUrlWithDisplayMode(window.location.href, "LIST");
+    });
+});
```

Comment:
Wrap in `(function () { ........ })();` to not have the `extendUrlWithDisplayMode` function accessible in the global scope. If that feature is wanted, we should define it as a function of the `sirius` object.